adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -241,33 +243,37 @@ private ConnectionLimitManager createConnectionLimitManager() {
 
     @Override
     protected void doStop(final Promise<Void> stopPromise) {
+        if (!stopCalled.compareAndSet(false, true)) {
+            log.trace("stop already called");
+            stopPromise.complete();
```

Comment:
FMPOV this should instead be tied to the outcome of the promise that had initially been used to shut down the adapter, e.g. by using an AtomicReference<Promise<Void>> instead ...

Reply:
I've amended the commit accordingly.